datafusion/expr/src/udaf.rs

```diff
@@ -442,7 +442,7 @@ pub trait AggregateUDFImpl: Debug + Send + Sync {
     /// not implement the method, returns an error. Order insensitive and hard
     /// requirement aggregators return `Ok(None)`.
     fn with_beneficial_ordering(
-        self: Arc<Self>,
+        &self,
```

Comment:
The `Arc<Self>` was hard to call from the aliased version. Is there a reason why it was Arc'd?

Reply:
I think the idea is that passing through the `&Arc` allows the result to be created without a deep copy., though maybe it is being overly optimized 🤔

Reply:
Hmm, dunno if there is a difference between passing an Arc<Self> and &self perf wise - neither should require copy. But I got it work with the Arc<> (by adding a .clone() there, but that should only clone the pointer and not the self so I think that's fine as well), so then I don't need to change this and that's probably safer 😅  

fixed in https://github.com/apache/datafusion/pull/12098/commits/33b89ceb3eb69e9b206121b883011bd85ea56b7e